### DISCUSSION TITLE
Left image in popupItem

### DISCUSSION THREAD

**Discussion by fury21** [LeoNatan/LNPopupController#537] [General] [answered]
Hi!
I want to put an album image on the left side of the popupItem, but I don't understand how to do it. I've tried different options, but they all don't work.

My image is large and the aspect ratio is not like a square, so I would like to reduce it to the size of 25x25 and make rounded corners, this requires access to the UIImageView, but I don't understand where it is?

How can this be done?
Thank you for your help!

What i want:
<img width="376" alt="Снимок экрана 2023-04-12 в 12 49 45" src="https://user-images.githubusercontent.com/30409500/231422157-ebc2d612-dc35-4559-8254-6a88abf96aaf.png">

If i do plyerManager.popupItem.image = image, i see this:
<img width="382" alt="Снимок экрана 2023-04-12 в 12 58 58" src="https://user-images.githubusercontent.com/30409500/231424290-64a96b15-3487-426a-86db-16884c8b87f7.png">

If i do plyerManager.popupItem.leadingBarButtonItems = [popupItem], i see this:
<img width="388" alt="Снимок экрана 2023-04-12 в 12 59 46" src="https://user-images.githubusercontent.com/30409500/231424455-e2d27c13-7550-43fc-8dcf-505741aa2290.png">

if i use image from example project with plyerManager.popupItem.leadingBarButtonItems = [popupItem], i see this (image on right side):
<img width="383" alt="Снимок экрана 2023-04-12 в 13 03 52" src="https://user-images.githubusercontent.com/30409500/231425538-0e228d96-9588-402d-aa2a-5d7b3db67f63.png">

**Comment by LeoNatan** [comment]
Hello,

You set the image of the popup item. 

If you want to modify how the image view displays the image, you have access to the image view here:
https://github.com/LeoNatan/LNPopupController/blob/0dfea7c146953f34ce3eb3d0793e90df17640891/LNPopupController/LNPopupController/LNPopupBar.h#L73

You can set the content mode to aspect fill, to have it be a square. By default, the image view is configured with round corners, so setting the content mode to aspect fill should give you the look you see in the first image.

**Comment by fury21** [comment]
I'm sorry, I didn't quite understand you and I don't know Obj-c language. 

I need to set the album image to the left side of the mini bar. What property do I need to pass the image to (UIImage) and then how do I access the UIImageView to adjust the rounding of the corners?

In my controller I have access to the properties popupItem, popupItem.leadingBarButtonItems, popupBar.ImageView

**Reply by LeoNatan** [reply]
If you didn’t understand what I wrote, I really cannot help you.

**Reply by fury21** [reply]
Actually you can) I need to understand which property to set the image to and how to configure ImageView afterwards?)
Thank u)

**Reply by iDevelopper** [reply]
Hello,

You set the image in the popupItem object in your playerManager for example (I assume this is your content view controller provided to the presentPopupBar method).

```swift
	popupItem.image = /* your image */
```

You configure your imageView with the popupBar (the popup bar that was presented with the container):
```swift
	containerVC.popupBar.imageView.contentMode = .scaleAspectFill
```

**Reply by fury21** [reply]
> ```swift
> containerVC.popupBar.imageView.contentMode = .scaleAspectFill
> ```

When I put an image in self.player manager.popupItem.image = image - everything is fine, the picture appears, but when I change self.player Manager.popupBar.ImageView.contentMode there is no effect, nothing happens. What am I missing?

**Reply by LeoNatan** [reply, marked as answer]
You are likely setting the content mode on the wrong popup bar. Set it for the bar of the controller **that is presenting**, not the one that is **being presented**.

**Reply by fury21** [reply]
Thank u! 
My problem was solved!
